### PR TITLE
Add gate object directly with QubitCircuit.add_gate(gate)

### PR DESCRIPTION
**Pull request by sahmed95**
This is in ref to #610. I have also cleaned up some pep8 and added a test for `add_gate`. This is now ready for review. @ajgpitch

### CODE REVIEW

File: qutip/qip/circuit.py [hunk 1]
```diff
@@ -172,15 +173,16 @@ def __init__(self, N, reverse_states=True):
...
+    def add_gate(self, gate, targets=None, controls=None, arg_value=None,
                  arg_label=None):
         """
         Adds a gate with specified parameters to the circuit.
 
         Parameters
         ----------
-        name: String
-            Gate name.
+        gate: String
```
Comment (ajgpitch) [inline qutip/qip/circuit.py]: maybe here you could say
```
    String or `Gate`
```
Not sure what the standard on this kind of thing is, but it is what I do

**Comment by ajgpitch** [comment]
I made one comment.
Happy to merge

**Comment by sahmed95** [comment]
This is ready for merge now.

**Comment by sahmed95** [comment]
Thank you.

**Pull request merged**